### ISSUE TITLE
F_C_STRING conflicts with intel oneAPI compliers

### ISSUE THREAD

**Issue by miicck** [libmbd/libmbd#65]
When compiling using the latest intel oneAPI compilers, this subroutine:

https://github.com/libmbd/libmbd/blob/1c432853bfe36312671bcd06c537c24a88559a2d/src/mbd_c_api.F90#L453

Conflicts with this function

https://www.intel.com/content/www/us/en/docs/fortran-compiler/developer-guide-reference/2024-1/f-c-string.html

I encountered this issue using the following compiler:
![image](https://github.com/libmbd/libmbd/assets/8690175/fa12c28a-87f9-4779-a2e2-0d6d91b00b96)

Whilst building the quantum espresso software. The following error was thrown:

![image](https://github.com/libmbd/libmbd/assets/8690175/1de5bf68-e5c1-4c6d-8be7-f96db200e6f3)

I'm not familiar enough with what this code is doing to know if this is a bug, or if it would be resolved with suitable compiler flags, so feel free to ignore or close this issue. I've renamed the subroutine in my local copy of libmbd as a workaround.

**Comment by aroudgar** [comment]
Yes, I got exactly the same error. A temporary workaround to fixes this issue would be to rename "f_c_string" to a different name within mbd_c_api.F90 file.

**Comment by xwuupb** [comment]
Please just fix it by simply renaming `f_c_string`! `f_c_string` is already in draft Fortran 2023 standard (https://j3-fortran.org/doc/year/23/23-007r1.pdf). Sooner or later no Fortran compiler in the world is able to compile this.

**Comment by Andrew-S-Rosen** [comment]
Fixed in https://github.com/libmbd/libmbd/pull/60
